### PR TITLE
Many sensor value for new Audi API

### PR DESCRIPTION
**Pull request by t0bias-r**

### CODE REVIEW

**Review by DivingUp — COMMENTED** [review]
adblue 2 times ind dashboard.py

**Comment by cdnninja** [comment]
This has lots of commits that are already reflected and merged. Could you rebase so it shows what is changing?

**Comment by cdnninja** [comment]
If you and @DivingUp want this merged though I can do that though.

**Comment by t0bias-r** [comment]
@cdnninja It's rebased

**Comment by cdnninja** [comment]
@DivingUp  let me know if good to merge. 
I'm good with it.

**Comment by DivingUp** [comment]
Yes, it Looks good

**Comment by cdnninja** [comment]
I created a merge conflict trying to get release working.   @t0bias-r  are you able to resolve conflict so I can merge? 

Previously I moved a release workflow over but I can't enable releases so that won't work.

File: custom_components/audiconnect/dashboard.py
Comment (Kolbi) [inline custom_components/audiconnect/dashboard.py:1]: Stupid question: What do you do with this file?

**Comment by t0bias-r** [comment]
Conflict resolved

**Comment by DivingUp** [comment]
we have to be careful with 
            #info = "shortterm"
            # await self.call_update(self.update_vehicle_shortterm, 3)
            # info = "longterm"
            # await self.call_update(self.update_vehicle_longterm, 3)
this 2 funtions in audi_connect_account.py...
this throws 403 by my car... to many 403 at one day -> audi will lock your account for 24h... so i've deactivated...

i had this problem before with the old audi_connect version... in the end my account (incl. APP) was completely blocked... car could no longer be opened via app and car key was inside... was not funny.

**Comment by Kolbi** [comment]
> we have to be careful with #info = "shortterm" # await self.call_update(self.update_vehicle_shortterm, 3) # info = "longterm" # await self.call_update(self.update_vehicle_longterm, 3) this 2 funtions in audi_connect_account.py... this throws 403 by my car... to many 403 at one day -> audi will lock your account for 24h... so i've deactivated...
> 
> **aiohttp.client_exceptions.ClientResponseError: 403, message='', url=URL('https://mal-1a.prd.ece.vwg-connect.com/api/bs/tripstatistics/v1/vehicles/WAUZXXXXXXXXX/tripdata/longTerm?type=list&from=1970-01-01T00:00:00Z&to=2024-03-05T08:38:22Z')**
> 
> i had this problem before with the old audi_connect version... in the end my account (incl. APP) was completely blocked... car could no longer be opened via app and car key was inside... was not funny.

The services are working for me with following URLs:
https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/tripstatistics/v1/Audi/CH/vehicles/[VIN]/tripdata/shortTerm
https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/tripstatistics/v1/Audi/CH/vehicles/[VIN]/tripdata/longTerm

Just: https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/tripstatistics/v1/Audi/CH/vehicles/[VIN]/tripdata/cyclic
gives an empty answer but 204.

If I will find time maybe I will adjust it.

**Comment by DivingUp** [comment]
> https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/tripstatistics/v1/Audi/CH/vehicles/[VIN]/tripdata/

aiohttp.client_exceptions.ClientResponseError: 401, message='', url=URL('https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/tripstatistics/v1/Audi/CH/vehicles/WAUZZZ4MXXX/tripdata/shortTerm?type=list&from=1970-01-01T00:00:00Z&to=2024-03-05T10:00:12Z

with self.vwToken & self._bearer_token_json  Token... alway 401... can you post the complete URL incl. Paramas and type of auth and headers?!

**Comment by Kolbi** [comment]
> update_vehicle_shortterm

Sadly I don't have time to test currently, but maybe it already works: https://github.com/Kolbi/audi_connect_ha

**Comment by t0bias-r** [comment]
I've captured the network traffic of Audi's Android App 4.23.1 and these URLs are used for trip statistics:

https://mal-3a.prd.eu.dp.vwg-connect.com/api/bs/tripstatistics/v1/vehicles/WAUZZZGE3RB987654/tripdata/longTerm?type=list&from=setreadingT00:00:00Z&to=2024-02-27T20:03:34Z
https://mal-3a.prd.eu.dp.vwg-connect.com/api/bs/tripstatistics/v1/vehicles/WAUZZZGE3RB987654/tripdata/shortTerm?type=list&from=setreadingT00:00:00Z&to=2024-02-27T20:03:34Z

Thr from parameter is different.
Maybe epoch 0 makes problems, too? In other time zones and/or countries?

**Comment by DivingUp** [comment]
> I've captured the network traffic of Audi's Android App 4.23.1 and these URLs are used for trip statistics:
> 
> https://mal-3a.prd.eu.dp.vwg-connect.com/api/bs/tripstatistics/v1/vehicles/WAUZZZGE3RB987654/tripdata/longTerm?type=list&from=setreadingT00:00:00Z&to=2024-02-27T20:03:34Z https://mal-3a.prd.eu.dp.vwg-connect.com/api/bs/tripstatistics/v1/vehicles/WAUZZZGE3RB987654/tripdata/shortTerm?type=list&from=setreadingT00:00:00Z&to=2024-02-27T20:03:34Z
> 
> Thr from parameter is different. Maybe epoch 0 makes problems, too? In other time zones and/or countries?

why now other urls.... old post :
https://fal-3a.prd.eu.dp.vwg-connect.com/fs-car/bs/tripstatistics/v1/Audi/CH/vehicles/[VIN]/tripdata/...

now:
https://mal-3a.prd.eu.dp.vwg-connect.com/api/bs/tripstatistics/v1/vehicles/WAUZZZGE3RB987654/tripdata/shortTerm?type=list&from=setreadingT00:00:00Z&to=2024-02-27T20:03:34Z

**Comment by t0bias-r** [comment]
@DivingUp I did not post an URL for Switzerland (ISO country code CH). What do you mean?

**Comment by DivingUp** [comment]
@t0bias-r sorry... i've not checked, that the last post ist from you, not from @Kolbi

**Comment by Kolbi** [comment]
> @t0bias-r sorry... i've not checked, that the last post ist from you, not from @Kolbi

My links were from https://github.com/cyr-ius/audiconnectpy and not captured. So maybe in the app they are using already newer ones. My links were working two days ago but I didn't managed to integrate them fastly to the solution here.

But maybe we should stick to the URLs from the app :)

**Comment by Kolbi** [comment]
I hope this PR will be integrated soon.

Other question: What is the dashboard.py file used for?

**Comment by cdnninja** [comment]
I think we still want merged. So going to do that.  If other tweaks needed can do another pr.

**Pull request merged**